packages/terra-date-input/src/DateInput.jsx

```diff
@@ -382,11 +382,12 @@ class DateInput extends React.Component {
     }
 
     const inputValue = event.target.value;
-    const maxValue = 9999;
+    const maxValue = 2100;
+    const minValue = 1900;
 
     // When 'Predictive text' is enabled on Android the maxLength attribute on the input is ignored so we have
     // to check the length of inputValue to make sure that it is less then 4.
-    if (inputValue.length > 5 || Number(inputValue) > maxValue) {
+    if (inputValue.length > 5 || Number(inputValue) > maxValue || (inputValue.length === 4 && Number(inputValue) < minValue)) {
```

Comment:
It allows entering `-` or `+` which it shouldn't. After that it messes up keyboard interaction where I can't delete and doesn't restrict the max length to be 4. Maybe [validNumericInput](https://github.com/cerner/terra-framework/blob/main/packages/terra-date-input/src/DateInputUtil.js#L8-L10) needs to also restrict `-` or `+` to prevent this from happening.

Reply:
Added here 6f09098